MonstieWash/Assets/Scripts/Video/VideoController.cs

```diff
@@ -0,0 +1,136 @@
+
+    //Private
+    private VideoPlayer m_vPlayer;
+    private float m_timeSinceLastInput; //How long since player moved mouse/joystick.
+    private bool m_moving; //Whether player is moving mouse/joystick currently.
+
+    private void OnEnable()
+    {
+        InputManager.Instance.OnMove += MovePerformed;
+        InputManager.Instance.OnMove_Ended += MoveEnded;
+    }
+
+    private void OnDisable()
+    {
+        InputManager.Instance.OnMove -= MovePerformed;
+        InputManager.Instance.OnMove_Ended -= MoveEnded;
+    }
+
+    /// <summary>
+    /// Reset move counter whenever input received.
+    /// </summary>
+    private void MovePerformed(Vector2 movementInput)
+    {
+        m_timeSinceLastInput = 0;
+        m_moving = true;
+    }
+
+    /// <summary>
+    /// //Stop video when moving stops.
+    /// </summary>
+    private void MoveEnded()
+    {
+        m_moving = false;
+        if (status == Status.Title) StopVideo();
+    }
+
+
+    private void Awake()
+    {
+        m_vPlayer = GetComponent<VideoPlayer>();
+
+        status = Status.Title;
+        if (playTitleOnLoad) PlayTitleVideo();
+    }
+
+    private void Update()
+    {
+        //Increment time since last input.
+        if(!m_moving) m_timeSinceLastInput += Time.deltaTime;
+
+        //See if video should play again.
+        if (status == Status.Title && playOnAfk && m_timeSinceLastInput > afkTimeToPlay && !m_vPlayer.isPlaying) PlayTitleVideo();
+
+    }
+
+    /// <summary>
+    /// Sets correct settings for title vid and plays.
+    /// </summary>
+    private void PlayTitleVideo()
+    {       
+        m_vPlayer.clip = TitleVideo;
+        m_vPlayer.isLooping = true;
+        m_vPlayer.enabled = true;
+        m_vPlayer.Play();
+    }
+
+    //Called by Slime poster navigation
+    /// <summary>
+    /// Sets correct settings and plays animatic.
+    /// </summary>
+    public void PlayAnimatic()
+    {      
+        m_vPlayer.clip = Animatic;
+        m_vPlayer.isLooping = false;
+        m_vPlayer.enabled = true;
+        m_vPlayer.Play();
+        status = Status.Animatic;
+        FindFirstObjectByType<SoundPlayer>().StopSound();
+    }
+
+    /// <summary>
+    /// Disables video player.
+    /// </summary>
+    private void StopVideo()
+    {
+        m_vPlayer.enabled = false;
+    }
+
+    /// <summary>
+    /// Sets correct settings after leaving main menu, assigns animatic finish function.
+    /// </summary>
+    //Called by main menu start game.
+    public void TransitionFromMainMenu()
+    {
+        StopVideo();
+        status = Status.Idle;
+        m_vPlayer.loopPointReached += OnVideoFinish;
+
+    }
+
+    /// <summary>
+    /// Transition to slime scene and delete video player.
+    /// </summary>
+    private void OnVideoFinish(VideoPlayer vp)
+    {
+        StopVideo();
+        status = Status.Idle;
+        GameSceneManager.Instance.StartNewLevel(levelFollowingAnimatic);
+        Destroy(this.gameObject);
+    }
```

Comment:
### ❗ **Requirement** | _This is preventing approval and must be addressed._

The only issue I could find was this function, where it destroys the video player once it starts the level. This breaks things if you try to play the level a second time since the video player is gone. Thankfully, an easy fix: just disable it again instead of destroying it. Since it's not actually doing anything while the game is going I think we can spare the memory.

```suggestion
    private void OnVideoFinish(VideoPlayer vp)
    {
        StopVideo();
        status = Status.Idle;
        GameSceneManager.Instance.StartNewLevel(levelFollowingAnimatic);
        m_vPlayer.enabled = false;
    }
```